modules/renter/contractor/contractor.go

```diff
@@ -152,7 +162,16 @@ func newContractor(cs consensusSet, w wallet, tp transactionPool, hdb hostDB, d
 		return nil, err
 	}
 
-	cs.ConsensusSetPersistentSubscribe(c, modules.ConsensusChangeID{})
+	err = cs.ConsensusSetPersistentSubscribe(c, c.lastChange)
+	if err == modules.ErrInvalidConsensusChangeID {
+		c.lastChange = modules.ConsensusChangeID{}
+		// ??? fix things ???
```

Comment:
the renter is going to be getting consensus set updates starting from the genesis block. So you need to reset any internal state in the renter that actively updates with the current block, so that the whole renter is consistent with a renter that's never seen any blocks before.

This is definitely something that you should be testing as well. It's worth breaking this logic into a separate `rescan()` function so that you can randomly call `rescan()` during your integration tests and make sure that nothing breaks when you do so.

Basically, have the expectation that the user could replace the consensus database at any time (well, it's probably safe to assume that they'll close `siad` first) and that this will trigger a rescan in the renter.